compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8140,6 +8141,9 @@ private BType checkOptionalFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr
             fieldAccessExpr.originalType = fieldAccessExpr.leafNode || !nillableExprType ? actualType :
                     types.getTypeWithoutNil(actualType);
         } else if (types.isLax(effectiveType)) {
+            if (effectiveType.tag == TypeTags.TYPEREFDESC) {
```

Comment:
We can use `Types.getReferredType()` method.

Reply:
I will update

Reply:
Updated. ff5f92a0e63a07b079b5be2d24af1d57d8a6d3cc